Android 40

FedorBysov opened the pull request:
ANDROID-40 Исправление бага навигации связанной с маршрутами + исправления бага в BottomNavigation

FedorBysov committed:
ANDROID-22 внедрил навигацию + 2 экрана с примером + Bottom bar

FedorBysov committed:
ANDROID-22 добавил NavigationRoutes

FedorBysov committed:
ANDROID-22 добавил Api для примера

FedorBysov committed:
ANDROID-22 переработка навигации + добавление комментариев

FedorBysov committed:
Merge branch 'develop' into ANDROID-22-

FedorBysov committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	core/ui/build.gradle.kts
#	core/ui/src/main/java/ru/yeahub/core_ui/theme/Typography.kt

FedorBysov committed:
Merge remote-tracking branch 'origin/develop' into develop

FedorBysov committed:
ANDROID-40 рефакторинг навигации + исправления бага в BottomNavigation +

FedorBysov committed:
Merge remote-tracking branch 'origin/develop' into develop

FedorBysov committed:
Merge branch 'develop' into ANDROID-40

# Conflicts:
#	core/navigation-impl/src/main/java/ru/yeahub/navigation_impl/AppNavigation.kt

FedorBysov committed:
ANDROID-40 Исправление бага навигации связанной с маршрутами + исправления бага в BottomNavigation

FedorBysov committed:
Merge remote-tracking branch 'origin/develop' into develop

FedorBysov committed:
Merge branch 'develop' into ANDROID-40

kish-dev commented on core/navigation-impl/src/main/java/ru/yeahub/navigation_impl/AppNavigation.kt:
```diff
@@ -61,79 +64,92 @@ fun AppNavigation(
+
+                            // Если мы в подмаршруте этого таба, навигируем на родительский
+                            if (currentRoute != null && currentRoute.startsWith(item.route)) {
+                                Log.d(
```
У нас же есть тимбер, надо, чтобы логи были в одном формате (посмотри на свой 1 файл)

kish-dev commented on core/navigation-impl/src/main/java/ru/yeahub/navigation_impl/AppNavigation.kt:
```diff
@@ -61,79 +64,92 @@ fun AppNavigation(
+                            if (currentRoute != null && currentRoute.startsWith(item.route)) {
+                                Log.d(
+                                    "NavDebug",
+                                    "Navigating to parent route: ${item.route} from child: $currentRoute"
```
Добавить в message название класса и функцию, где этот лог

kish-dev commented on core/navigation-impl/src/main/java/ru/yeahub/navigation_impl/AppNavigation.kt:
```diff
@@ -61,79 +64,92 @@ fun AppNavigation(
+                    FeatureRoute.StubFeature.FEATURE_NAME
+                },
+                modifier = Modifier,
+//                анимация
```
Раскомментировать или удалить

kish-dev commented on feature/example-home/impl/src/main/java/ru/yeahub/example_home/impl/HomeFeatureImpl.kt:
```diff
@@ -0,0 +1,54 @@
+                        userId = userId,
+                        userName = userName
+                    )
+                    Log.d("NavDebug", "Navigating to profile: $profilePath")
```
На тимбер и название класса + функции

kish-dev commented on feature/example-home/impl/src/main/java/ru/yeahub/example_home/impl/HomeFeatureImpl.kt:
```diff
@@ -0,0 +1,54 @@
+                    val questionsPath = FeatureRoute.QuestionsFeature.createQuestionsPath(
+                        parentRoute = currentHomeRoute
+                    )
+                    Log.d("NavDebug", "Navigating to questions: $questionsPath")
```
аналогично

kish-dev commented on feature/example-home/impl/src/main/java/ru/yeahub/example_home/impl/HomeFeatureModule.kt:
```diff
@@ -1,8 +1,18 @@
-} 
\ No newline at end of file
+    single<HomeScreenApi> {
+        Log.d("NavDebug", "Creating HomeScreenApiImpl")
```
аналогично

kish-dev commented on feature/example-home/impl/src/main/java/ru/yeahub/example_home/impl/HomeFeatureModule.kt:
```diff
@@ -1,8 +1,18 @@
+        HomeScreenApiImpl()
+    }
+    single { 
+        Log.d("NavDebug", "Creating HomeFeatureImpl")
```
аналогично

kish-dev commented on feature/example-home/impl/src/main/java/ru/yeahub/example_home/impl/HomeScreenApiImpl.kt:
```diff
@@ -59,51 +50,51 @@ class HomeScreenApiImpl : HomeScreenApi {
-            }
-        }
-    }
+//    @Composable
```
удалить закомментированный код

kish-dev commented on feature/example-profile/impl/src/main/java/ru/yeahub/example_profile/impl/ProfileFeatureImpl.kt:
```diff
@@ -0,0 +1,54 @@
+        parentRoute: String,
+        modifier: Modifier
+    ) {
+        Log.d("NavDebug", "registerGraph() of ProfileFeatureImpl with parentRoute: $parentRoute")
```
аналогично

kish-dev commented on feature/example-profile/impl/src/main/java/ru/yeahub/example_profile/impl/ProfileFeatureImpl.kt:
```diff
@@ -0,0 +1,54 @@
+            val userId = backStackEntry.arguments?.getString("userId") ?: ""
+            val userName = backStackEntry.arguments?.getString("userName") ?: ""
+            
+            Log.d("NavDebug", "ProfileScreen: userId=$userId, userName=$userName")
```
аналогично

kish-dev commented on feature/example-profile/impl/src/main/java/ru/yeahub/example_profile/impl/ProfileFeatureImpl.kt:
```diff
@@ -0,0 +1,54 @@
+                userName = userName,
+                parentRoute = parentRoute,
+                onBackClick = {
+                    Log.d("NavDebug", "Navigating back from profile")
```
аналогично

kish-dev commented on feature/example-questions/api/src/main/java/com/example/api/navigation/QuestionsRoutes.kt:
```diff
@@ -1,12 +1,10 @@
-
-    fun questionsRoute() = BASE_ROUTE
+object QuestionsRoutes  {
+//    private const val BASE_ROUTE = "questions"
```
удалить закомментированный код

kish-dev commented on feature/example-questions/impl/src/main/java/com/example/impl/QuestionsFeatureImpl.kt:
```diff
@@ -0,0 +1,38 @@
+        parentRoute: String,
+        modifier: Modifier
+    ) {
+        Log.d("NavDebug", "registerGraph() of QuestionsFeatureImpl with parentRoute: $parentRoute")
```
аналогично

kish-dev commented on feature/example-questions/impl/src/main/java/com/example/impl/QuestionsFeatureImpl.kt:
```diff
@@ -0,0 +1,38 @@
+        navGraphBuilder.composable(currentQuestionsRoute) {
+            questionScreen.QuestionsScreen(
+                onBackClick = {
+                    Log.d("NavDebug", "Navigating back from questions")
```
аналогично

FedorBysov committed:
ANDROID-40 Исправление логов

FedorBysov commented:
Исправил логи

FedorBysov committed:
ANDROID-40
	•	Исправление ошибок глубокой навигации
	•	Проведен эксперемент с навигацией через уведомления

FedorBysov committed:
ANDROID-40 рефакторинг

FedorBysov commented:
https://github.com/user-attachments/assets/7aee412d-97fc-4031-b5f8-39f7977b882f

kish-dev committed:
Merge branch 'develop' into ANDROID-40
